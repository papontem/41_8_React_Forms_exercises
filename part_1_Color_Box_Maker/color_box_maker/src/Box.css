.Box-container {
	border: 1px solid gold;
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: center;
}
.Box {
	
	outline: 1px solid red;
}
