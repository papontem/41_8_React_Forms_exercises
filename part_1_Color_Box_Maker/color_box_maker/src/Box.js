/*
    - Displays a div with a background color, width, and height based on the props.
    - Includes a button with the text "X" next to each box.
    - Utilizes a function passed down as a prop to remove the specific box when the "X" button is clicked.
*/

function Box(props) {
	const { width, height, backgroundColor, deleteBox } = props;

	// console.log("width:", width);
	// console.log("height:", height);
	// console.log("backgroundColor:", backgroundColor);

	const boxStyle = {
		width: width || "100px",
		height: height || "100px",
		backgroundColor: backgroundColor || "lightblue",
	};

	const handleDeleteClick = () => {
		deleteBox();
	};

	return (
		<div className="Box-container">
			<div className="Box" style={boxStyle}>
				{/* I AM A BOOOOOOOOX */}
			</div>
			<button onClick={handleDeleteClick}>X</button>
		</div>
	);
}

export default Box;
