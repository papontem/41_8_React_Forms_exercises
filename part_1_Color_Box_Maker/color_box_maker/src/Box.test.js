import { render} from '@testing-library/react';
import Box from './Box';

test('renders box without crashing', () => {
  render(<Box />);
});