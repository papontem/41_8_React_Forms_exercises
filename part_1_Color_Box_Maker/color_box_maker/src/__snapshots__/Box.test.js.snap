// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      style="width: 100px; height: 100px; background-color: lightblue;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
